### ISSUE TITLE
selector-pseudo-element-no-unknown should skip custom selector by default

### ISSUE THREAD

**Issue by MoOx** [stylelint/stylelint#2146]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

feature

> Which rule, if any, is this issue related to?

selector-pseudo-element-no-unknown

> What CSS is needed to reproduce this issue?

This is the tricky part. Using the CSS standalone does not seems to allow reproduction

https://github.com/MoOx/phenomic/blob/84d0ad5c0a0b66f3f060ec7c592b2a04e1e4a178/docs/src/index.global.css

To reproduce:

```console
git clone https://github.com/MoOx/phenomic.git
cd phenomic
npm install # grab a coffee
cd docs
npm test
```

> What stylelint configuration is needed to reproduce this issue?

```json
  "stylelint": {
    "extends": "stylelint-config-standard",
    "rules": {
      "block-no-empty": null,
      "comment-empty-line-before": null,
      "comment-whitespace-inside": null
    }
}
```

https://github.com/MoOx/phenomic/blob/84d0ad5c0a0b66f3f060ec7c592b2a04e1e4a178/docs/package.json#L148-L155

> Which version of stylelint are you using?

"stylelint": "^7.2.0",
"stylelint-config-standard": "^13.0.2",

But reproducible with 7.6 and 15.0

> How are you running stylelint: CLI, PostCSS plugin, Node API?

`stylelint "src/**/*.css"`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
src/index.global.css
 97:1  ✖  Unexpected unknown pseudo-class selector ":--headings"   selector-pseudo-class-no-unknown
```

**Comment by evilebottnawi** [comment]
@MoOx Thanks for feedback. Duplicate https://github.com/stylelint/stylelint/issues/2141
